ESQL: DS: investigate flakiness in ExternalDistributedStressIT tests

### Description

These tests [fail sometimes](https://buildkite.com/elastic/elasticsearch-pull-request/builds/134261#019d4d77-d768-4c64-8c19-576744528adb), but not reproducibly:
```
REPRODUCE WITH: ./gradlew ":x-pack:plugin:esql:qa:server:multi-node:javaRestTest" --tests "org.elasticsearch.xpack.esql.qa.multi_node.ExternalDistributedStressIT.testHeterogeneousSplitSizes" -Dtests.seed=9DE8FAC556CF043C -Dtests.locale=be-Cyrl-BY -Dtests.timezone=Asia/Kabul -Druntime.java=25
--
 
ExternalDistributedStressIT > testHeterogeneousSplitSizes FAILED
java.lang.AssertionError: Row count mismatch for mode coordinator_only expected:<4143> but was:<3876>
at __randomizedtesting.SeedInfo.seed([9DE8FAC556CF043C:17702B22172D41ED]:0)
at org.junit.Assert.fail(Assert.java:89)
at org.junit.Assert.failNotEquals(Assert.java:835)
at org.junit.Assert.assertEquals(Assert.java:647)
at org.elasticsearch.xpack.esql.qa.multi_node.ExternalDistributedStressIT.testHeterogeneousSplitSizes(ExternalDistributedStressIT.java:169)
 
REPRODUCE WITH: ./gradlew ":x-pack:plugin:esql:qa:server:multi-node:javaRestTest" --tests "org.elasticsearch.xpack.esql.qa.multi_node.ExternalDistributedStressIT.testManyUniformSplitsWithAggregation" -Dtests.seed=9DE8FAC556CF043C -Dtests.locale=be-Cyrl-BY -Dtests.timezone=Asia/Kabul -Druntime.java=25
 
ExternalDistributedStressIT > testManyUniformSplitsWithAggregation FAILED
java.lang.NullPointerException: Cannot invoke "java.lang.Number.longValue()" because the return value of "java.util.List.get(int)" is null
at __randomizedtesting.SeedInfo.seed([9DE8FAC556CF043C:87F2B9A2E01E89B4]:0)
at org.elasticsearch.xpack.esql.qa.multi_node.ExternalDistributedStressIT.testManyUniformSplitsWithAggregation(ExternalDistributedStressIT.java:122)
 
REPRODUCE WITH: ./gradlew ":x-pack:plugin:esql:qa:server:multi-node:javaRestTest" --tests "org.elasticsearch.xpack.esql.qa.multi_node.ExternalDistributedStressIT.testManyUniformSplitsWithTopN" -Dtests.seed=9DE8FAC556CF043C -Dtests.locale=be-Cyrl-BY -Dtests.timezone=Asia/Kabul -Druntime.java=25
 
ExternalDistributedStressIT > testManyUniformSplitsWithTopN FAILED
java.lang.AssertionError: Top-N value mismatch at index 0 for mode coordinator_only expected:<2515> but was:<2413>
at __randomizedtesting.SeedInfo.seed([9DE8FAC556CF043C:7AF7797E9A2BC6BE]:0)
at org.junit.Assert.fail(Assert.java:89)
at org.junit.Assert.failNotEquals(Assert.java:835)
at org.junit.Assert.assertEquals(Assert.java:647)
at org.elasticsearch.xpack.esql.qa.multi_node.ExternalDistributedStressIT.testManyUniformSplitsWithTopN(ExternalDistributedStressIT.java:147)
```

Or from [another failure](https://buildkite.com/elastic/elasticsearch-pull-request/builds/134277#019d4d9d-6feb-4a49-a54e-deb38079b092):
```
REPRODUCE WITH: ./gradlew ":x-pack:plugin:esql:qa:server:multi-node:javaRestTest" --tests "org.elasticsearch.xpack.esql.qa.multi_node.ExternalDistributedStressIT.testHeterogeneousSplitSizes" -Dtests.seed=19A981B6E605FE64 -Dtests.locale=ar-LB -Dtests.timezone=US/East-Indiana -Druntime.java=25
--
 
ExternalDistributedStressIT > testHeterogeneousSplitSizes FAILED
java.lang.AssertionError: Row count mismatch for mode round_robin expected:<1888> but was:<0>
at __randomizedtesting.SeedInfo.seed([19A981B6E605FE64:93315051A7E7BBB5]:0)
at org.junit.Assert.fail(Assert.java:89)
at org.junit.Assert.failNotEquals(Assert.java:835)
at org.junit.Assert.assertEquals(Assert.java:647)
at org.elasticsearch.xpack.esql.qa.multi_node.ExternalDistributedStressIT.testHeterogeneousSplitSizes(ExternalDistributedStressIT.java:169)
 
REPRODUCE WITH: ./gradlew ":x-pack:plugin:esql:qa:server:multi-node:javaRestTest" --tests "org.elasticsearch.xpack.esql.qa.multi_node.ExternalDistributedStressIT.testManyUniformSplits" -Dtests.seed=19A981B6E605FE64 -Dtests.locale=ar-LB -Dtests.timezone=US/East-Indiana -Druntime.java=25
 
ExternalDistributedStressIT > testManyUniformSplits FAILED
java.lang.AssertionError: Row count mismatch for mode round_robin expected:<2385> but was:<0>
at __randomizedtesting.SeedInfo.seed([19A981B6E605FE64:2D17977D6F3007CC]:0)
at org.junit.Assert.fail(Assert.java:89)
at org.junit.Assert.failNotEquals(Assert.java:835)
at org.junit.Assert.assertEquals(Assert.java:647)
at org.elasticsearch.xpack.esql.qa.multi_node.ExternalDistributedStressIT.testManyUniformSplits(ExternalDistributedStressIT.java:103)


```

--- a/x-pack/plugin/esql/qa/server/multi-node/src/javaRestTest/java/org/elasticsearch/xpack/esql/qa/multi_node/ExternalDistributedStressIT.java
+++ b/x-pack/plugin/esql/qa/server/multi-node/src/javaRestTest/java/org/elasticsearch/xpack/esql/qa/multi_node/ExternalDistributedStressIT.java
@@ -12,12 +12,14 @@ import com.carrotsearch.randomizedtesting.annotations.ThreadLeakFilters;
 import org.elasticsearch.common.Strings;
 import org.elasticsearch.core.SuppressForbidden;
 import org.elasticsearch.test.TestClustersThreadFilter;
+import org.junit.Before;
 
 import java.nio.charset.StandardCharsets;
 import java.util.Iterator;
 import java.util.List;
 import java.util.Locale;
 import java.util.Map;
+import java.util.UUID;
 
 import static org.elasticsearch.xpack.esql.datasources.S3FixtureUtils.BUCKET;
 import static org.elasticsearch.xpack.esql.datasources.S3FixtureUtils.WAREHOUSE;
@@ -30,17 +32,28 @@ import static org.elasticsearch.xpack.esql.datasources.S3FixtureUtils.addBlobToF
  * content for formula-based assertions across all distribution modes.
  * <p>
  * All synthetic data is held entirely in-memory via the S3 fixture's blob map.
+ * <p>
+ * Each test method uses a unique sub-prefix under {@code warehouse/stress/} so the
+ * coordinator's external source listing cache (keyed by glob path) cannot return a
+ * stale {@code FileList} from a previous method after blobs were cleared and
+ * repopulated (see elasticsearch#145559).
  */
 @SuppressForbidden(reason = "uses S3 fixture handler for direct blob population")
 @ThreadLeakFilters(filters = { TestClustersThreadFilter.class })
 public class ExternalDistributedStressIT extends AbstractExternalDistributedIT {
 
-    private static final String STRESS_PREFIX = WAREHOUSE + "/stress/";
+    /** Per-test run prefix; includes a random segment so listing cache keys differ each method. */
+    private String stressRunPrefix;
+
+    @Before
+    public void initStressRunPrefix() {
+        stressRunPrefix = WAREHOUSE + "/stress/" + UUID.randomUUID() + "/";
+    }
 
     @org.junit.After
     public void clearStressSplits() {
         var handler = s3Fixture.getHandler();
-        String blobPrefix = Strings.format("/%s/%s", BUCKET, STRESS_PREFIX);
+        String blobPrefix = Strings.format("/%s/%s", BUCKET, stressRunPrefix);
         Iterator<String> it = handler.blobs().keySet().iterator();
         while (it.hasNext()) {
             if (it.next().startsWith(blobPrefix)) {
@@ -62,7 +75,7 @@ public class ExternalDistributedStressIT extends AbstractExternalDistributedIT {
     private void populateStressSplits(int numSplits, int rowsPerSplit) {
         var handler = s3Fixture.getHandler();
         for (int i = 0; i < numSplits; i++) {
-            String key = String.format(Locale.ROOT, "%spart-%05d.csv", STRESS_PREFIX, i);
+            String key = String.format(Locale.ROOT, "%spart-%05d.csv", stressRunPrefix, i);
             addBlobToFixture(handler, key, generateCsvSplit(i, rowsPerSplit));
         }
     }
@@ -79,13 +92,13 @@ public class ExternalDistributedStressIT extends AbstractExternalDistributedIT {
                 sb.append(i).append(",").append(rowId).append(",").append(value).append("\n");
             }
             globalOffset += numRows;
-            String key = String.format(Locale.ROOT, "%spart-%05d.csv", STRESS_PREFIX, i);
+            String key = String.format(Locale.ROOT, "%spart-%05d.csv", stressRunPrefix, i);
             addBlobToFixture(handler, key, sb.toString().getBytes(StandardCharsets.UTF_8));
         }
     }
 
     private String stressQuery(String suffix) {
-        return externalS3Query(STRESS_PREFIX + "*.csv") + suffix;
+        return externalS3Query(stressRunPrefix + "*.csv") + suffix;
     }
 
     public void testManyUniformSplits() throws Exception {

